services/workflows-service/src/alert/alert.service.intg.test.ts

```diff
@@ -82,6 +84,90 @@ describe('AlertService', () => {
         .transactionDate(faker.date.recent(6));
     });
 
+    describe('Rule: DORMANT', () => {
+      let alertDefinition: AlertDefinition;
+
+      const execWithTransactionFactory = async (
```

Comment:
You preferred reusability over readability in the tests you implemented.
In tests it should be exactly the opposite. Tests should be highly maintainable and easy to read. Code duplications and readability are preferred over reusability.

This function makes the tests less readable. I think it would be better if you write this code inside the test instead of in a reusable function.

Reply:
There is a small bug that i could find his root cause, that's why i recreate transaction factory 

made this code more readable and added types as well